### DISCUSSION TITLE
Support Hermes on tvOS

### DISCUSSION THREAD

**Discussion by miblanchard** [react-native-tvos/react-native-tvos#334] [Feature Requests]
Hey @douglowder, I see in the docs that hermes is currently not available for tvOS.  I'm curious how deep you've investigated what it would take to add that support?  I'll admit I haven't looked into what went into adding iOS support to hermes, though I recall a strong series of info from callstack about this.

I'm interested in anything you are able to share.

Thanks again.

**Comment by douglowder** [comment]
Callstack did the work to support Hermes on iOS architectures, but not tvOS (which is arm64). I'm sure it can be done but it would be a significant effort. @grabbou might be able to add more detail on what needs to be done. Good reference to start: https://www.callstack.com/blog/bringing-hermes-to-ios-in-react-native

**Comment by YazeedAsaad** [comment]
hello @douglowder  just out curiosity is hermes being worked on for tvos or not yet ?

**Comment by douglowder** [comment]
I am now actively working on this.

**Comment by douglowder** [comment]
Progress: Hermes now successfully builds on Apple TV, see my changes at https://github.com/douglowder/hermes/commits/tvos

Integration into React Native will also require that I port libevent to Apple TV. Working on that.

**Comment by geroale** [comment]
Hello @douglowder, any update on this?

**Comment by douglowder** [comment]
Yes @geroale there is now a 0.70.3 branch, and I've made some progress. Pod install now works, but there are still issues with getting the RNTester build to complete successfully.

**Comment by geroale** [comment]
Hello @douglowder, when compiling in release mode on tvOS, the app is not able to run the js bundle because it seems to use hermes, even if it's not enabled in the podspec for tvOS. Hermes must be disabled also on the iOS project.

Apart from this problem, are there any news regarding support ad hermes?

Thank you in advance

**Comment by douglowder** [comment]
@geroale can the problem with Hermes in release mode be reproduced with a new project?

I am actively working on Hermes for tvOS. Currently targeting 0.71 for releasing it, as there are going to be changes in how Hermes is packaged for that release.

**Comment by geroale** [comment]
@douglowder I confirm that a similar/related issue is present also on a fresh project created with [template](https://github.com/react-native-tvos/react-native-template-typescript-tv).

The debug scheme is compiling and running fine.
The release one is compiling too but is crashing at startup with `com.facebook.react.JavaScript (9): "Could not get BatchedBridge, make sure your bundle is packaged correctly"` at [this line](https://github.com/react-native-tvos/react-native-tvos/blob/a8775346269cdbb3e530e7130fb1d3ea31e261f4/React/CxxBridge/RCTCxxBridge.mm#L335)

To reproduce the issue in the template project, it is only needed to change the Podfile with `:hermes_enabled => true` on the iOS target and `:hermes_enabled => false` on tvOS target.

It is not exactly the issue we are having on our project but probably because we are using [react-native-navigation](https://github.com/wix/react-native-navigation) which handles the bridge differently I think. But the issue is still related with the bundle: the bundle is generated with hermes even if it's not enabled in the podspec for tvOS. Hermes must be disabled also on the iOS target for us.

**Reply by douglowder** [reply]
@geroale I'm looking at this issue, and confirm the issue happens with a fresh project.

Not sure if there is a good fix yet, as Hermes and JSC bundles are not compatible and iOS and tvOS use the same bundle. I'll investigate and see if I can come up with a fix or workaround.

**Reply by geroale** [reply]
@douglowder Can this be solved with the support of Hermes on tvOS?

**Reply by douglowder** [reply]
Once that is working, you would be able to have both iOS and tvOS on Hermes, yes.

**Comment by geroale** [comment]
Hello @douglowder, any update on this? Can we help in any way?

**Reply by douglowder** [reply]
The current tentative plan:

- Complete a few more bug fixes and release them as 0.69.6-1
- Do a 0.70.x release without Hermes for tvOS (January 2023)
- Do the 0.71.x release with Hermes for tvOS (February or March 2023)

**Reply by geroale** [reply]
Hello @douglowder, do you have any news about 0.70.x and 0.71.x? Can we help in some way?

**Comment by douglowder** [comment]
A prerelease with Hermes working on tvOS is now available. https://github.com/react-native-tvos/react-native-tvos/discussions/507

**Comment by douglowder** [comment]
Hermes for tvOS is released in 0.71.10-0.
